pybamm/models/submodels/interface/li_plating/irreversible_plating.py

```diff
@@ -0,0 +1,79 @@
+#
+# Class for irreversible Li plating
+#
+import pybamm
+from .base_plating import BasePlating
+
+
+class IrreversiblePlating(BasePlating):
+    """Base class for irreversible Li plating.
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    domain : str
+        The domain of the model either 'Negative' or 'Positive'
+    **Extends:** :class:`pybamm.li_plating.BasePlating`
+    """
+
+    def __init__(self, param, domain):
+        super().__init__(param, domain)
+
+    def get_fundamental_variables(self):
+        c_plated_Li = pybamm.Variable(
+            "Plated Li concentration",
+            domain=self.domain.lower() + " electrode",
+            auxiliary_domains={"secondary": "current collector"},
+        )
+
+        variables = self._get_standard_concentration_variables(c_plated_Li)
+
+        return variables
+
+    def get_coupled_variables(self, variables):
+        param = self.param
+        phi_s_n = variables[f"{self.domain} electrode potential"]
+        phi_e_n = variables[f"{self.domain} electrolyte potential"]
+        c_e_n = variables[f"{self.domain} electrolyte concentration"]
+        C_plating = param.C_plating
+        phi_ref = param.U_n_ref / param.potential_scale
+
+        if f"{self.domain} electrode sei film overpotential" in variables:
+            eta_sei = variables[f"{self.domain} electrode sei film overpotential"]
+        else:
+            eta_sei = pybamm.Scalar(0)
```

Comment:
Ideally the sei film overpotential should always be defined, sometimes to zero. Otherwise, if the submodels are called in the wrong order, the sei film overpotential might be non-zero but this would miss it